### PR TITLE
[SEDONA-411] Add RS_Rotation

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-411. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Add RS_Rotation
- Add RS_GeoTransform

## How was this patch tested?

- Passed new tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

**Comment by jiayuasu** [comment]
@furqaankhan Is the result identical to PostGIS?

**Comment by furqaankhan** [comment]
Yes it is.

**Comment by furqaankhan** [comment]
@jiayuasu I want to get Kristin's opinion before merging, so don't merge yet.

File: common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java [hunk 1]
```diff
@@ -127,6 +128,70 @@ public static String getGeoReference(GridCoverage2D raster, String format) {
...
+
+        // Rotation
+        thetaI = Math.acos(scaleX / magnitudeI);
+        double thetaTest = Math.acos(skewY / magnitudeI);
+        if (thetaTest < Math.PI / 2) {
+            thetaI = -thetaI;
+        }
+
+        // Angular separation
+        thetaIJ = Math.acos((((scaleX * skewX) + (skewY * scaleY)) / (magnitudeI * magnitudeJ)));
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java:173]: can `magnitude` be zero or other invalid values? If so, we need to add handlers and test cases.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java [hunk 2]
```diff
@@ -127,6 +128,70 @@ public static String getGeoReference(GridCoverage2D raster, String format) {
...
+        // angle from transformed i axis to transformed j axis (Radians positive counter-clockwise)
+        double thetaIJ;
+
+        RenderedImage renderedImage = raster.getRenderedImage();
+
+        // x ordinate of the upper-left corner of the upper-left pixel
+        double offsetX = renderedImage.getTileGridXOffset();
+
+        // y ordinate of the upper-left corner of the upper-left pixel
+        double offsetY = renderedImage.getTileGridYOffset();
```
Comment (Kontinuation) [inline common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java:151]: This is not correct. `offsetX` and `offsetY` should be directly retrieved from the upperLeftX and upperLeftY components of the raster metadata.

File: common/src/test/java/org/apache/sedona/common/raster/RasterAccessorsTest.java [hunk 3]
```diff
@@ -60,6 +59,22 @@ public void testSrid() throws FactoryException {
...
+        double actual = RasterAccessors.getRotation(emptyRaster);
+        double expected = -1.4711276743037347;
+        assertEquals(expected, actual, 1e-9);
+    }
+
+    @Test
+    public void testGeoTransform() throws FactoryException {
+        GridCoverage2D emptyRaster = RasterConstructors.makeEmptyRaster(1, 10, 15, 1, 2, 1, -1, 10, 10, 0);
+        double[] actual = RasterAccessors.getGeoTransform(emptyRaster);
+        double[] expected = new double[] {10.04987562112089, 10.04987562112089, -1.4711276743037347, -1.5707963267948966, 0.0, 0.0};
```
Comment (Kontinuation) [inline common/src/test/java/org/apache/sedona/common/raster/RasterAccessorsTest.java:74]: We are getting 0 offsets for rasters with non-zero offsets, this is caused by the problem mentioned above.

**Review by Kontinuation — CHANGES_REQUESTED** [review]

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**